crates/nu-protocol/src/engine/pwd_per_drive.rs

```diff
@@ -0,0 +1,289 @@
+                base.push(&path_str[2..]); // Join PWD with path parts after "C:"
+                return Some(base);
+            }
+        }
+        None
+    }
+
+    /// Implementation for maintainer and fs_client
+    /// Windows env var for drive
+    /// essential for integration with windows native shell CMD/PowerShell
+    /// and the core mechanism for supporting PWD-per-drive with nushell's
+    /// powerful layered environment system.
+    /// Returns uppercased "=X:".
+    fn env_var_for_drive(drive_letter: char) -> String {
+        let drive_letter = drive_letter.to_ascii_uppercase();
+        format!("={}:", drive_letter)
+    }
+
+    /// get pwd for drive:
+    /// 1. From env_var, if no,
+    /// 2. From sys_absolute, if no,
+    /// 3. Construct root path to drives
+    fn get_pwd_on_drive(stack: &Stack, engine_state: &EngineState, drive_letter: char) -> String {
+        let env_var_for_drive = env_var_for_drive(drive_letter);
+        let mut abs_pwd: Option<String> = None;
+        if let Some(pwd) = stack.get_env_var(engine_state, &env_var_for_drive) {
+            if let Ok(pwd_string) = pwd.clone().into_string() {
+                abs_pwd = Some(pwd_string);
+            }
+        }
+        if abs_pwd.is_none() {
+            if let Some(sys_pwd) = get_full_path_name_w(&format!("{}:", drive_letter)) {
+                abs_pwd = Some(sys_pwd);
+            }
+        }
+        if let Some(pwd) = abs_pwd {
+            ensure_trailing_delimiter(&pwd)
+        } else {
+            format!(r"{}:\", drive_letter)
+        }
+    }
+
+    /// Check if input path is relative path for drive letter,
+    /// which should be expanded with PWD-per-drive.
+    /// Returns Some(drive_letter) or None, drive_letter is upper case.
+    fn need_expand(path: &str) -> Option<char> {
+        let chars: Vec<char> = path.chars().collect();
+        if chars.len() == 2 || (chars.len() > 2 && chars[2] != '/' && chars[2] != '\\') {
+            extract_drive_letter(path)
+        } else {
+            None
+        }
+    }
+
+    /// Extract the drive letter from a path, return uppercased
+    /// drive letter or None
+    fn extract_drive_letter(path: &str) -> Option<char> {
+        let chars: Vec<char> = path.chars().collect();
+        if chars.len() >= 2 && chars[0].is_ascii_alphabetic() && chars[1] == ':' {
+            Some(chars[0].to_ascii_uppercase())
+        } else {
+            None
+        }
+    }
+
+    /// Ensure a path has a trailing `\\` or '/'
+    fn ensure_trailing_delimiter(path: &str) -> String {
+        if !path.ends_with('\\') && !path.ends_with('/') {
+            format!(r"{}\", path)
+        } else {
+            path.to_string()
+        }
+    }
+
+    /// get_full_path_name_w
+    /// Call Windows system API (via omnipath crate) to expand
+    /// absolute path
+    fn get_full_path_name_w(path_str: &str) -> Option<String> {
+        use omnipath::sys_absolute;
+        use std::path::Path;
```

Comment:
can we put these two use statements at the top please, just so we know what's being used at a glance?

Reply:
OK

Reply:
I see my comments are resolved but I don't see any new commits. Did I miss them or have you just not pushed them yet?